code/datums/beam.dm

```diff
@@ -169,7 +169,7 @@
 		if(!living)
 			xenos -= living
 			continue
-		if(living.stat == DEAD)
+		if(living.stat == DEAD || (living.status_flags & INCORPOREAL) || (living.status_flags & GODMODE))
```

Comment:
Actually one change - you don't need the two checks separate. they're both status_flags so you can check them at the same time, since we want to pull them out if they have either.